src/pages/ProductList/index.jsx

```diff
@@ -0,0 +1,55 @@
+import React, { useState, useEffect } from 'react';
+import products from '../../assets/data/products';
+import Pagination from '../../components/Pagination';
+import ProductsGrid from '../../components/ProductsGrid';
+import Sidebar from '../../components/Sidebar';
+import useFilters from '../../utils/hooks/useFilters';
+import { StyledProductList, StyledTitle } from './styled';
+
+function ProductList() {
+  const [isLoading, setIsLoading] = useState(true);
+
+  useEffect(() => {
+    const loadingTimeout = setTimeout(() => {
+      setIsLoading(false);
+    }, 2000);
+    return () => {
+      clearTimeout(loadingTimeout);
+    };
+  }, []);
+
+  const {
+    filter: categoriesFilter,
+    handleFilterChange: handleCategoriesFilterChange,
+  } = useFilters([]);
```

Comment:
I think you can refactor the `useFilters` hook so you can pass the products and categories from the `assets`, then you receive the filtered products (the products with active filters) and the mapped categories (this now contains the `checked` property) also receive the `handleCategoriesFilterChange` function.

```jsx
const {
    filteredProducts,
    mappedCategories,
    handleFilterChange: handleCategoriesFilterChange,
  } = useFilters(productCategories.results, products.results);
```

So every time, when the `handleCategoriesFilterChange` function is called on clicking on a category, the arrays `filteredProducts` and `mappedCategories` will re-create. This refactoring has many benefits due to you are removing the logic from components.

Reply:
The `mappedCategories` looks like this:
```js
{
id:"....",
data: {
    name: 'Bed & Bath',
    ...
},
active: false
}
```
The `active` property is the new one, that indicates which categories are enabled.